cxotime/cxotime.py

```diff
@@ -174,6 +175,65 @@ def __init__(self, *args, **kwargs):
 
         super(CxoTime, self).__init__(*args, **kwargs)
 
+    # Make a classmethod that replaces utils.get_range_in_chunks
+    @classmethod
+    def linspace(
+        cls,
+        start: CxoTimeLike,
+        stop: CxoTimeLike,
+        num: int | None = None,
+        step_max: u.Quantity | None = None,
+    ):
+        # This method will use the linspace method if num is defined
+        # or the step_max method if step_max is defined
+        """
+        Get uniform time chunks for a given time range.
+
+        Output times either divide the time range into ``num`` chunks or are
+        uniformly spaced by up to ``step_max``, and cover the time
+        range from ``start`` to ``stop``.
+
+        Parameters
+        ----------
+        start : CxoTimeLike
+            Start time of the time range.
+        stop : CxoTimeLike
+            Stop time of the time range.
+        num : int | None
+            Number of time bins.
+        step_max : u.Quantity (timelike)
+            Maximum time interval for each chunk.
+
+        Returns
+        -------
+        CxoTime
+            CxoTime with time bin edges for each chunk.
+        """
+        start = CxoTime(start)
+        stop = CxoTime(stop)
+
+        if num is not None and step_max is not None:
+            raise ValueError("Only one of num or step_max can be defined")
+
+        if num is not None:
+            if num <= 0:
+                raise ValueError("num must be positive nonzero")
+            times = np.linspace(start, stop, num + 1)
+        else:
+            # Require that step_max is a positive nonzero quantity
```

Comment:
This `else` block only needs to compute the correct `num`, and then that can be passed to a common `times = np.linspace(start, stop, num + 1)` outside of the if / else.